htsohm/htsohm.py

```diff
@@ -198,125 +195,16 @@ def run_all_simulations(material):
-
-    elif 'gas_adsorption_0' in simulations and 'gas_adsorption_1' in simulations:
-        if config['artificial_data'] == 'off':
-            arguments = [material.run_id, material]
-            if 'helium_void_fraction' in simulations:
-                arguments.append(material.vf_helium_void_fraction)
-            results = simulation.gas_adsorption_0.run(*arguments)
-            material.update_from_dict(results)
-            results = simulation.gas_adsorption_1.run(*arguments)
-            material.update_from_dict(results)
-            material.gas_adsorption_bin = calc_bin(
-                abs(material.ga0_absolute_volumetric_loading - material.ga1_absolute_volumetric_loading),
-                *config['gas_adsorption_0']['limits'],
-                config['number_of_convergence_bins']
-            )
-        else:
-            results = {}
-            results['ga0_absolute_volumetric_loading'] = material.artificial_gas_adsorption()
-            results['ga1_absolute_volumetric_loading'] = material.artificial_gas_adsorption()
-            material.update_from_dict(results)
-            material.gas_adsorption_bin = calc_bin(
-                material.ga0_absolute_volumetric_loading,
-                *config['gas_adsorption_0']['limits'],
-                config['number_of_convergence_bins']
-            )
-            print('gas adsorption\t|  {}\t|  {}\t      |'.format(
-                round(results['ga0_absolute_volumetric_loading'], 4),
-                material.gas_adsorption_bin))
-
-    elif 'gas_adsorption_0' not in simulations:
-        material.gas_adsorption_bin = 0
-    ############################################################################
-    # run surface area simulation
-    if 'surface_area' in simulations:
-        if config['artificial_data'] == 'off':
-            results = simulation.surface_area.run(
-                    material.run_id, material)
-            material.update_from_dict(results)
-            material.surface_area_bin = calc_bin(
-                material.sa_volumetric_surface_area,
-                *config['surface_area']['limits'],
-                config['number_of_convergence_bins']
-            )
-        else:
-            results = {}
-            results['sa_volumetric_surface_area'] = material.artificial_surface_area()
-            material.update_from_dict(results)
-            material.surface_area_bin = calc_bin(
-                material.sa_volumetric_surface_area,
-                *config['surface_area']['limits'],
-                config['number_of_convergence_bins']
-            )
-            print('surface area\t|  {}\t|  {}\t      |'.format(
-                round(results['sa_volumetric_surface_area'], 4),
-                material.surface_area_bin))
-
-    else:
-        material.surface_area_bin = 0
+    simulation_types = config['simulations']
+    for simulation_type in simulation_types:
+        results = get_simulation(simulation_type).run(material)
```

Comment:
likely would be more robust to pass the config to the simulation. That way you can more easily handle multiple tests of a certain type.